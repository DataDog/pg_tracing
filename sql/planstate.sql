
-- Test with explain_spans disabled
SET pg_tracing.explain_spans = false;
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000001-0000000000000001-01'*/ SELECT s.relation_size + s.index_size
FROM (SELECT
      pg_relation_size(C.oid) as relation_size,
      pg_indexes_size(C.oid) as index_size
    FROM pg_class C) as s limit 1;
SELECT span_type, span_operation, deparse_info FROM peek_ordered_spans where trace_id='00000000000000000000000000000001';

-- Test with explain_spans enabled
SET pg_tracing.explain_spans = true;
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000002-0000000000000002-01'*/  SELECT s.relation_size + s.index_size
FROM (SELECT
      pg_relation_size(C.oid) as relation_size,
      pg_indexes_size(C.oid) as index_size
    FROM pg_class C) as s limit 1;
SELECT span_type, span_operation, deparse_info FROM peek_ordered_spans where trace_id='00000000000000000000000000000002';

-- Check generated spans when deparse is disabled
SET pg_tracing.deparse_plan=false;
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000003-0000000000000003-01'*/ SELECT * from pg_tracing_test where a=1;
SELECT span_operation, deparse_info, parameters, lvl from peek_ordered_spans where trace_id='00000000000000000000000000000003';

-- Check generated spans when deparse is enabled
SET pg_tracing.deparse_plan=true;
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000004-0000000000000004-01'*/ SELECT * from pg_tracing_test where a=1;
SELECT span_operation, deparse_info, parameters, lvl from peek_ordered_spans where trace_id='00000000000000000000000000000004';

-- Clean created spans
CALL clean_spans();
