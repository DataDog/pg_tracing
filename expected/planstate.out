-- Test with planstate_spans disabled
SET pg_tracing.planstate_spans = false;
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000001-0000000000000001-01'*/ SELECT s.relation_size + s.index_size as sum_size
FROM (SELECT
      pg_relation_size(C.oid) as relation_size,
      pg_indexes_size(C.oid) as index_size
    FROM pg_class C) as s limit 1 \gset
SELECT span_type, span_operation, deparse_info FROM peek_ordered_spans where trace_id='00000000000000000000000000000001';
  span_type   |                                                       span_operation                                                       | deparse_info 
--------------+----------------------------------------------------------------------------------------------------------------------------+--------------
 Select query | SELECT s.relation_size + s.index_size as sum_size                                                                         +| 
              | FROM (SELECT pg_relation_size(C.oid) as relation_size, pg_indexes_size(C.oid) as index_size FROM pg_class C) as s limit $1 | 
 Planner      | Planner                                                                                                                    | 
 ExecutorRun  | ExecutorRun                                                                                                                | 
(3 rows)

-- Test with planstate_spans enabled
SET pg_tracing.planstate_spans = true;
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000002-0000000000000002-01'*/  SELECT s.relation_size + s.index_size as sum_size
FROM (SELECT
      pg_relation_size(C.oid) as relation_size,
      pg_indexes_size(C.oid) as index_size
    FROM pg_class C) as s limit 1 \gset
SELECT span_type, span_operation, deparse_info FROM peek_ordered_spans where trace_id='00000000000000000000000000000002';
  span_type   |                                                       span_operation                                                       | deparse_info 
--------------+----------------------------------------------------------------------------------------------------------------------------+--------------
 Select query | SELECT s.relation_size + s.index_size as sum_size                                                                         +| 
              | FROM (SELECT pg_relation_size(C.oid) as relation_size, pg_indexes_size(C.oid) as index_size FROM pg_class C) as s limit $1 | 
 Planner      | Planner                                                                                                                    | 
 ExecutorRun  | ExecutorRun                                                                                                                | 
 Limit        | Limit                                                                                                                      | 
 SubqueryScan | SubqueryScan on s                                                                                                          | 
 SeqScan      | SeqScan on pg_class c                                                                                                      | 
(6 rows)

-- Check generated spans when deparse is disabled
SET pg_tracing.deparse_plan=false;
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000003-0000000000000003-01'*/ SELECT * from pg_tracing_test where a=1;
 a |          b           
---+----------------------
 1 | aaa                 
(1 row)

SELECT span_operation, deparse_info, parameters, lvl from peek_ordered_spans where trace_id='00000000000000000000000000000003';
                   span_operation                    | deparse_info | parameters | lvl 
-----------------------------------------------------+--------------+------------+-----
 SELECT * from pg_tracing_test where a=$1;           |              | $1 = 1     |   1
 Planner                                             |              |            |   2
 ExecutorRun                                         |              |            |   2
 IndexScan using pg_tracing_index on pg_tracing_test |              |            |   3
(4 rows)

-- Check generated spans when deparse is enabled
SET pg_tracing.deparse_plan=true;
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000004-0000000000000004-01'*/ SELECT * from pg_tracing_test where a=1;
 a |          b           
---+----------------------
 1 | aaa                 
(1 row)

SELECT span_operation, deparse_info, parameters, lvl from peek_ordered_spans where trace_id='00000000000000000000000000000004';
                   span_operation                    |    deparse_info     | parameters | lvl 
-----------------------------------------------------+---------------------+------------+-----
 SELECT * from pg_tracing_test where a=$1;           |                     | $1 = 1     |   1
 Planner                                             |                     |            |   2
 ExecutorRun                                         |                     |            |   2
 IndexScan using pg_tracing_index on pg_tracing_test | Index Cond: (a = 1) |            |   3
(4 rows)

-- Clean created spans
CALL clean_spans();
